core/operator.go

```diff
@@ -129,6 +164,45 @@ func (o *Operator) Seal() error {
 	if err := o.chain.db.CommitTx(); err != nil {
 		log.WithError(err).Error("failed to commit db tx")
 		//TODO: need recover here
+		err := o.chain.db.RollbackTx()
+		if err != nil {
+			return err
+		}
+		return err
+	}
+
+	for _, tx := range block.Transactions() {
+		hash := tx.Hash()
+		o.txPool.removeTx(hash, true)
+	}
+
+	return nil
+}
+
+// SealDeposit get txs from txpool and construct block, then seal the block
+func (o *Operator) SealDeposit(txs types.Transactions) error {
+	block := o.constructBlock(txs)
+	newDbTx := o.chain.db.BeginTx()
+	if !newDbTx {
+		return errors.New("database race detected, there should be no tx pending when plasma operator commit a block")
+	}
+	// append block to chain and update chain head
+	if err := o.chain.WriteDepositBlock(block); err != nil {
+		err := o.chain.db.RollbackTx()
+		if err != nil {
+			return err
+		}
+
+		return err
+	}
+	o.chain.ReplaceHead(block)
+	if err := o.chain.db.CommitTx(); err != nil {
```

Comment:
new `utxo` generated by deposit block should bet put into the utxo set, before `commitTx`